File: Emiliano_3er_sem/Python-3er-sem/Leccion14/mundo_pc/teclado.py
from mundo_pc.dispositivo_entrada import Dispositivoentrada

class Teclado(Dispositivoentrada):
    contador_tecleado = 0
    def __init__(self, marca, tipoentrada):
        Tecleado.contador_tecleado += 1
        self._id_tecleado = Teclado.contador_tecleado
        super().__init__(marca, tipoentrada)
        
    def __str__(self):
        return f"Id: {self._id_tecleado}, Marca: {self._marca}, Tipo Entrada: {self._tipo_entrada}"

if __name__ == "__main__":
    teclado1 = Teclado("HP","USB")
    print(teclado1)
    teclado2 = Teclado("Gamer","USB")
    print(teclado2)

File: Emiliano_3er_sem/Java_3er_sem/Leccion01/Mundo_pc/Teclado.java
package Emiliano_3er_sem.Java_3er_sem.Leccion01.Mundo_pc;

public class Teclado  extends DispositivoEntrada{
    private final int idTeclado;
    private static int contadorTeclados;

    public Teclado(String tipoEntrada, String marca){
        super(tipoEntrada, marca);
        this.idTeclado = ++Teclado.contadorTeclados;
    }

    @Override
    public String toString(){
        return "Teclado{" + "IdTeclado=" + idTeclado + ", " + super.toString()+"}";
    }
}
